To: vim_dev@googlegroups.com
Subject: Patch 7.3.099
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.099
Problem:    Crash when splitting a window with zero height. (Yukihiro
	    Nakadaira)
Solution:   Don't set the fraction in a window with zero height.
Files:	    src/window.c


*** ../vim-7.3.098/src/window.c	2010-12-17 17:35:05.000000000 +0100
--- src/window.c	2011-01-08 14:41:32.000000000 +0100
***************
*** 986,992 ****
  
      /* Set w_fraction now so that the cursor keeps the same relative
       * vertical position. */
!     set_fraction(oldwin);
      wp->w_fraction = oldwin->w_fraction;
  
  #ifdef FEAT_VERTSPLIT
--- 986,993 ----
  
      /* Set w_fraction now so that the cursor keeps the same relative
       * vertical position. */
!     if (oldwin->w_height > 0)
! 	set_fraction(oldwin);
      wp->w_fraction = oldwin->w_fraction;
  
  #ifdef FEAT_VERTSPLIT
*** ../vim-7.3.098/src/version.c	2011-01-04 19:25:46.000000000 +0100
--- src/version.c	2011-01-08 14:44:02.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     99,
  /**/

-- 
From "know your smileys":
 8-O 	"Omigod!!" (done "rm -rf *" ?)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
